pkg/cluster/prepare.go

```diff
@@ -0,0 +1,130 @@
+package cluster
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/fluxcd/pkg/untar"
+	"github.com/weaveworks/pctl/pkg/runner"
+)
+
+const (
+	kubectlCmd = "kubectl"
+)
+
+// Fetcher will download a manifest tar file from a remote repository.
+type Fetcher struct {
+	Client *http.Client
+}
+
+// Applier applies the previously generated manifest files.
+type Applier struct {
+	Runner runner.Runner
+}
+
+// Preparer will prepare an environment.
+type Preparer struct {
+	PrepConfig
+	Fetcher *Fetcher
+	Applier *Applier
+}
+
+// PrepConfig defines configuration options for prepare.
+type PrepConfig struct {
+	// BaseURL is given even one would like to download manifests from a fork
+	// or a test repo.
+	BaseURL     string
+	Location    string
+	Version     string
+	KubeContext string
+	KubeConfig  string
+	DryRun      bool
+}
+
+// NewPreparer creates a preparer with set dependencies ready to be used.
+func NewPreparer(cfg PrepConfig) (*Preparer, error) {
+	tmp, err := ioutil.TempDir("", "pctl-manifests")
+	if err != nil {
+		return nil, fmt.Errorf("failed to create temp folder for manifest files: %w", err)
+	}
+	cfg.Location = tmp
+	return &Preparer{
+		PrepConfig: cfg,
+		Fetcher: &Fetcher{
+			Client: http.DefaultClient,
+		},
+		Applier: &Applier{
+			Runner: &runner.CLIRunner{},
+		},
+	}, nil
+}
+
+// Prepare will prepare an environment with everything that is needed to run profiles.
+func (p *Preparer) Prepare() error {
+	if _, err := os.Stat(p.Location); err != nil {
+		return err
+	}
```

Comment:
_possibly_ not necessary now, since it is no longer configurable, but 🤷‍♀️

Reply:
We wanna make sure temp is accessible, yes?

Reply:
wouldn't 51 have failed if it wasn't tho?

Reply:
meh, either way extra security is nbd

Reply:
Possibly it would have.. 🤔

Reply:
yeh it looks like `ioutil.Tempdir` does this:
```go
 try := filepath.Join(dir, prefix+nextRandom()+suffix)
 err = os.Mkdir(try, 0700)
 if os.IsExist(err) {
```
dir being os.TempDir in this case

Reply:
:D Excellent. :)